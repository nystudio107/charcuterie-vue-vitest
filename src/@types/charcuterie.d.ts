
interface CharcuterieItem {
    name: string;
    type: CharcuterieItemType;
    calories: number;
}
