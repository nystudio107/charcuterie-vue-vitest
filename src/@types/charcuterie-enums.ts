export enum CharcuterieItemType {
    Cheese = "cheese",
    Meat = "meat",
    Fruit = "fruit",
    Vegetable = "vegetable",
}
