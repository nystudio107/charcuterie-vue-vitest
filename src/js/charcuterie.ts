export enum CharcuterieItemType {
  Cheese,
  Meat,
  Fruit,
}

const charcuterieBoard: CharcuterieItem[] = [];

export function getAllItems() {
  return charcuterieBoard;
}

export function getItemsOfType(type: CharcuterieItemType) {
  return charcuterieBoard.filter((value:CharcuterieItem) => value.type === type)
}

export function addCharcuterieItem(item: CharcuterieItem) {
  charcuterieBoard.push(item);
}
