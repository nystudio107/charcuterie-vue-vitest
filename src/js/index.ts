import { createApp } from 'vue'
import '../css/style.css'
import App from '../vue/App.vue'

createApp(App).mount('#app')
