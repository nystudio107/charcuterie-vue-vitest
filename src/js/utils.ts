export const kebabCase = (string: string) => string;
