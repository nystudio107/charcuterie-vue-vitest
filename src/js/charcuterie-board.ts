import {CharcuterieItemType} from "../@types/charcuterie-enums";

const charcuterieBoard: CharcuterieItem[] = [];

export function getAllItems() {
  return charcuterieBoard;
}

export function clearAllItems() {
  charcuterieBoard.length = 0;
}
export function getItemsOfType(type: CharcuterieItemType): CharcuterieItem[] {
  return charcuterieBoard.filter((value: CharcuterieItem) => value.type === type)
}

export function addItem(item: CharcuterieItem|undefined) {
  if (typeof item !== 'undefined') {
    charcuterieBoard.push(item);
  }
}
