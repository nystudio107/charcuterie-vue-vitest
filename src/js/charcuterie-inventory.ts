import {CharcuterieItemType} from '@/@types/charcuterie-enums';

const charcuterieInventory: CharcuterieItem[]  = [
  {
    name: 'salami',
    type: CharcuterieItemType.Meat,
    calories: 200,
  },
  {
    name: 'olives',
    type: CharcuterieItemType.Fruit,
    calories: 100
  }
];

export function getInventory() {
  return charcuterieInventory;
}

export function getInventoryItem(name: string): CharcuterieItem|undefined {
  return charcuterieInventory.find((value: CharcuterieItem) => value.name === name);
}
