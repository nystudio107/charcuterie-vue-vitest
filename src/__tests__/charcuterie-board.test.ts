import {CharcuterieItemType} from "../@types/charcuterie-enums";
import {getAllItems, getItemsOfType, addItem} from '../js/charcuterie-board';
import {getInventoryItem} from "../js/charcuterie-inventory";
import {it, expect, describe} from 'vitest';

describe("charcuterie-board.ts", () => {
  it('should add items to the charcuterie board', () => {
    addItem(getInventoryItem('olives'));
    addItem(getInventoryItem('salami'));
    expect(getAllItems()).toMatchSnapshot();
  });

  it('should handle adding undefined items', () => {
    addItem(undefined);
    expect(getAllItems()).toMatchSnapshot();
  });

  it ('should return the proper types of items', () => {
    addItem(getInventoryItem('olives'));
    addItem(getInventoryItem('salami'));
    const fruits = getItemsOfType(CharcuterieItemType.Fruit);
    fruits.forEach((value: CharcuterieItem) => {
      expect(value.type).eq(CharcuterieItemType.Fruit);
    });
  });
});
