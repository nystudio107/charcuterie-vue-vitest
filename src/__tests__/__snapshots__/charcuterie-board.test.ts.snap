// Vitest Snapshot v1

exports[`charcuterie-board.ts > should add items to the charcuterie board 1`] = `
[
  {
    "calories": 100,
    "name": "olives",
    "type": "fruit",
  },
  {
    "calories": 200,
    "name": "salami",
    "type": "meat",
  },
]
`;

exports[`charcuterie-board.ts > should handle adding undefined items 1`] = `[]`;
